.nuspec/Microsoft.Maui.Controls.SingleProject.targets

```diff
@@ -63,6 +64,7 @@
     <Compile
         Condition=" '$(TargetPlatformIdentifier)' == 'windows' and '$(WindowsProjectFolder)' != '' "
         Include="$(WindowsProjectFolder)**/*$(DefaultLanguageSourceExtension)" />
+    <None Include="@(_MauiCompileBeforeRemove)" Exclude="@(Compile)" />
```

Comment:
Can you try putting this `<None/>` block in target like:
```xml
<Target Name="_AddFilesToNone" BeforeTargets="CoreCompile">
  <ItemGroup>
    <None Include="@(_MauiCompileBeforeRemove)" Exclude="@(Compile)" />
    <!-- You might not even need this -->
    <_MauiCompileBeforeRemove Remove="@(_MauiCompileBeforeRemove)" />
  </ItemGroup>
</Target>
```
And see what the IDE does?

Reply:
The IDE does not seem to notice the task. I suppose it wouldn't really because we add things to Compile on compile, but that doesn't run when the IDE stands around.